transport/http.go

```diff
@@ -73,37 +72,71 @@ var (
+	// ELASTIC_APM_API_KEY environment variable.
 	APIKey string
 
-	// SecretToken holds the secret token configured in the APM Server.
+	// SecretToken holds the secret token configured in the APM Server, used
+	// for authenticating the agent.
+	//
+	// If unspecified, SecretToken will be initialized using the
+	// ELASTIC_APM_SECRET_TOKEN envirohnment variable.
 	SecretToken string
 
-	// ServerCert can be set if you have configured your APM Server with a
-	// self signed TLS certificate, or you want to verify the server certificate
-	// matches this exact TLS certificate.
+	// ServerCert holds the path to a PEM-encoded TLS certificate that must
+	// match the APM Server-supplied certificate. This can be used to pin a
+	// self signed certificate. If this is set, then SkipServerVerify will be
+	// ignored.
+	//
+	// If unspecified, ServerCert will be initialized using the
+	// ELASTIC_APM_SERVER_CERT environment variable.
 	ServerCert string
 
-	// ServerCACert can be set if you want to specify a path to the PEM-encoded
-	// Certificate Authority to verify the Server TLS certificate.
+	// ServerCACert holds the path to a PEM-encoded Certificate Authority
+	// certificate that will be used for verifying the server's TLS certificate
+	// chain.
+	//
+	// If unspecified, ServerCACert will be initialized using the
+	// ELASTIC_APM_SERVER_CA_CERT_FILE environment variable.
 	ServerCACert string
 
 	// ServerURLs holds the URLs for your Elastic APM Server. The Server
 	// supports both HTTP and HTTPS. If you use HTTPS, then you may need to
 	// configure your client machines so that the server certificate can be
 	// verified. You can disable certificate verification with SkipServerVerify.
-	// If no URL is specified, then the transport will use the default URL
-	// "http://localhost:8200".
+	//
+	// If no URLs are specified, then ServerURLs will be initialized using the
+	// ELASTIC_APM_SERVER_URL environment variable, defaulting to
+	// "http://localhost:8200" if the environment variable is not set.
 	ServerURLs []*url.URL
 
 	// ServerTimeout holds the timeout for requests made to your Elastic APM
-	// server. When set to zero, it will default to 30 seconds. Negative values
+	// server.
+	//
+	// When set to zero, it will default to 30 seconds. Negative values
 	// are not allowed.
+	//
+	// If ServerTimeout is zero, then it will be initialized using the
+	// ELASTIC_APM_SERVER_TIMEOUT environment variable, defaulting to
+	// 30 seconds if the environment variable is not set. Negative values are
+	// not allowed, and will cause NewHTTPTransport to return an error.
 	ServerTimeout time.Duration
 
-	// SkipServerVerify skips TLS certificate validation of the APM Server.
+	// SkipServerVerify skips TLS certificate verification of the APM Server.
+	// By default, the server's TLS certificate will be verified.
+	//
+	// If false, SkipServerVerify will be initialized using the
+	// ELASTIC_APM_VERIFY_SERVER_CERT environment variable.
```

Comment:
The behavior I'm used to (and I think what we generally do in apm-server / apm-agent-go) is that an explicitly configured value in the constructor takes precedence over an environment variable. I would expect that setting `false` explicitly for `SkipServerVerify` would ensure the value is false, which isn't the case (the environment variable can overrule an explicitly set `false`). That behavior is clearly described in the comment, so that's fine, but I'm wondering if this is something we want to consider.

Reply:
@marclop also raised this in standup yesterday.

What do you both think about removing `ServerCert`, `ServerCACert`, and `SkipVerifyCert` fields, and instead introducing a `TLS *tls.Config` field? Only if `TLS` field is nil would `ELASTIC_APM_SERVER_CERT`, `ELASTIC_APM_SERVER_CA_CERT_FILE`, and `ELASTIC_APM_VERIFY_SERVER_CERT` be used.

That would make the very simplest things a little bit more verbose, but provides more flexibility to users for more complicated use cases.

Reply:
sounds goo to me!

Reply:
:+1: that sounds like a good solution to me